src/main/java/appeng/tile/misc/TileCondenser.java

```diff
@@ -182,63 +169,14 @@ public double getRequiredPower()
 	}
 
 	@Override
-	public IInventory getInternalInventory()
-	{
-		return this.inv;
-	}
-
-	@Override
-	public void setInventorySlotContents( final int i, final ItemStack itemstack )
-	{
-		if( i == 0 )
-		{
-			if( !itemstack.isEmpty() )
-			{
-				this.addPower( itemstack.getCount() );
-			}
-		}
-		else
-		{
-			this.inv.setInventorySlotContents( 1, itemstack );
-		}
-	}
-
-	@Override
-	public boolean isItemValidForSlot( final int i, final ItemStack itemstack )
+	public IItemHandlerModifiable getInternalInventory()
 	{
-		return i == 0;
+		return this.combinedInv;
 	}
 
 	@Override
-	public void onChangeInventory( final IInventory inv, final int slot, final InvOperation mc, final ItemStack removed, final ItemStack added )
+	public void onChangeInventory( final IItemHandlerModifiable inv, final int slot, final InvOperation mc, final ItemStack removed, final ItemStack added )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)